colcon_sanitizer_reports/event_handlers/sanitizer_report.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2019 Open Source Robotics Foundation
+# Licensed under the Apache License, version 2.0
+
+from colcon_core.event.job import JobEnded
+from colcon_core.event_handler import EventHandlerExtensionPoint
+from colcon_core.location import get_log_path
+from colcon_core.plugin_system import satisfies_version
+from colcon_output.event_handler.log import STDOUT_STDERR_LOG_FILENAME
+from colcon_sanitizer_reports.sanitizer_log_parser import SanitizerLogParser
+
+
+class SanitizerReportEventHandler(EventHandlerExtensionPoint):
+    """Generate a report of all Sanitizer ERRORs and WARNINGs."""
+
+    ENABLED_BY_DEFAULT = False
+
+    def __init__(self):
+        super().__init__()
+        satisfies_version(EventHandlerExtensionPoint.EXTENSION_POINT_VERSION, '^1.0')
+        self.enabled = False
+        self._log_parser = SanitizerLogParser()
+
+    def __call__(self, event):
+        data = event[0]
+
+        if isinstance(data, JobEnded):
+            self._handle(event)
+
+    def _handle(self, event):
+        job = event[1]
+        log_f = get_log_path() / job.identifier / STDOUT_STDERR_LOG_FILENAME
+        if not log_f.exists():
```

Comment:
It's not recommended in Python to do this. Instead we should try to open the file catch the exception if needed.

https://docs.quantifiedcode.com/python-anti-patterns/readability/asking_for_permission_instead_of_forgiveness_when_working_with_files.html

Reply:
I tend to favor an "exceptions should be exceptional" / "no exceptions should be expected" approach.

I will not claim to be an expert in python performance but I feel like creating exceptions which bundle loads of execution context that we don't need and are going to discard is wasteful.

Reply:
@nuclearsandwich I think the point of this idiom is to avoid race conditions. E.g. you test if the file exist, someone removes the file, you try to open the file and get an exception you never thought you could get.

Reply:
I can yield to that reasoning.

Reply:
Agree, changed this into a `try...catch` in the next commit.